### PR TITLE
Replace fakeweb with webmock

### PR DESCRIPTION
**Pull request by malept**
fakeweb hasn't been updated since 2010. Meanwhile, webmock has been fairly actively maintained (the last release was last month). The switchover was fairly trivial.

### CODE REVIEW

File: test/helper.rb [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
-
-FakeWeb.allow_net_connect = %r[^https?://127.0.0.1]
+require 'webmock/minitest'
 
 module MiniTest
   class Test
     def setup
-      FakeWeb.clean_registry
+      WebMock.reset!
+      WebMock.disable_net_connect!(allow_localhost: true)
```
Comment (matthuhiggins) [inline test/helper.rb]: This belongs outside of setup

Comment (matthuhiggins) [inline test/helper.rb]: The `disable_net_connect`, that is

File: test/elastic_record/log_subscriber_test.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ def setup
   # end
 
   def test_request_notification
-    FakeWeb.register_uri(:any, %r[/test], status: ["200", "OK"], body: Oj.dump('the' => 'response'))
+    stub_request(:any, "#{Widget.elastic_connection.servers.first}/test").to_return(status: 200, body: Oj.dump('the' => 'response'))
```
Comment (matthuhiggins) [inline test/elastic_record/log_subscriber_test.rb]: Can this use `stub_es_request`?

Comment (malept) [inline test/elastic_record/log_subscriber_test.rb]: I'd have to genericize `stub_es_request` to take in an ES connection object and then move it to `MiniTest::Test`, but sure.

Comment (matthuhiggins) [inline test/elastic_record/log_subscriber_test.rb]: Nevermind

Comment (ebarendt) [inline test/elastic_record/log_subscriber_test.rb]: Why? To use `Widget.elastic_connection.servers.first`? Webmock can just test against `/test` like the original did

Comment (ebarendt) [inline test/elastic_record/log_subscriber_test.rb]: (no familiarity with stub_es_request)

Comment (malept) [inline test/elastic_record/log_subscriber_test.rb]: webmock needs the hostname, unlike fakeweb. I *could* hardcode the hostname, but that feels wrong.

Comment (ebarendt) [inline test/elastic_record/log_subscriber_test.rb]: https://github.com/data-axle/content_system/blob/28af3f376eb07098f96b63bbee4415980c28039a/test/models/feedback/api_update_test.rb#L74-L74

Comment (malept) [inline test/elastic_record/log_subscriber_test.rb]: Huh. The webmock documentation led me to believe that the hostname was mandatory. I will fix.

File: test/helper.rb (hunk 1 above)
Comment (malept) [inline test/helper.rb]: Fixed in https://github.com/data-axle/elastic_record/pull/46/commits/a67637c1ee3ed832294e8bacf9b83209678d1c49

File: test/elastic_record/log_subscriber_test.rb (hunk 2 above)
Comment (malept) [inline test/elastic_record/log_subscriber_test.rb]: Fixed in https://github.com/data-axle/elastic_record/pull/46/commits/137508b3f002f8eb2fa378e9c3a264a227cc30b4 - I tried removing `stub_es_request` entirely but that passed through all of the requests, which is what we don't want.

**Pull request merged**